src/ManifoldMuseum.jl

```diff
@@ -1,30 +1,39 @@
 module ManifoldMuseum
 
-export Manifold
+import Base: isapprox, exp, log
+import LinearAlgebra: dot, norm
+
 
 abstract type Manifold end
 
-retract!(M::Manifold, y, x, v) = exp!(M,y,x,v)
-retract(M::Manifold, x, v) = retract!(M,copy(x),x,v)
+isapprox(m::Manifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+
+retract!(M::Manifold, y, x, v) = exp!(M, y, x, v)
+retract(M::Manifold, x, v) = retract!(M, copy(x), x, v)
 
 project_tangent!(M::Manifold, w, x, v) = error("Not implemented")
-project_tangent(M::Manifold, x, v) = project_tangent!(M,copy(x),x,v)
+project_tangent(M::Manifold, x, v) = project_tangent!(M, copy(x), x, v)
 
 distance(M::Manifold, x, y) = error("Not implemented")
 
-dot(M::Manifold, x, v, w) = dot(v,w)
+dot(M::Manifold, x, v, w) = dot(v, w)
 norm(M::Manifold, x, v) = sqrt(dot(M, x, v, v))
 
 exp!(M::Manifold, y, x, v) = error("Not implemented")
-exp(M::Manifold, x, v) = exp!(M,copy(x),x,v)
+exp(M::Manifold, x, v) = exp!(M,copy(x), x, v)
 
 log!(M::Manifold, v, x, y) = error("Not implemented")
-log(M::Manifold, x, y) = error("Not implemented")
+
+function log(M::Manifold, x, y)
+    v = zero_tangent_vector(M, x)
```

Comment:
Perhaps better to declare as undefined instead of filling with zeros that are then replaced.

Reply:
But then we'd need an `undefined_tangent_vector`? That seems overkill

Reply:
True, was thinking something like `TVector(M, x, undef)`, but since not all vectors can be instantiated with `undef`, I can see why we wouldn't do that.